package/base-files/files/sbin/sysupgrade

```diff
@@ -118,32 +118,41 @@ list_conffiles() {
 		BEGIN { conffiles = 0 }
 		/^Conffiles:/ { conffiles = 1; next }
 		!/^ / { conffiles = 0; next }
-		conffiles == 1 { print }
+		# in format: <space><filename><space><hash>
+		conffiles == 1 { print $(NF) "  " substr($0,2,length($0)-length($(NF))-2) }
 	' /usr/lib/opkg/status
 }
 
 list_changed_conffiles() {
-	# Cannot handle spaces in filenames - but opkg cannot either...
-	list_conffiles | while read file csum; do
+	list_conffiles | while read -r csum_file; do
+		file="${csum_file#*  }"
 		[ -r "$file" ] || continue
 
-		echo "${csum}  ${file}" | busybox sha256sum -sc - || echo "$file"
+		echo "${csum_file}" | busybox sha256sum -sc - || echo "$file"
 	done
 }
 
-list_static_conffiles() {
-	local filter=$1
+find_bkpfiles() {
+	find "$@" \( -type f -o -type l \) \
+		${SKIP_UNCHANGED:+\( \! -exec test -e "/rom/{}" \; -o \! -exec cmp -s "/{}" "/rom/{}" \; \)} \
+		-print
+}
 
-	find $(sed -ne '/^[[:space:]]*$/d; /^#/d; p' \
-		/etc/sysupgrade.conf /lib/upgrade/keep.d/* 2>/dev/null) \
-		\( -type f -o -type l \) $filter 2>/dev/null
+list_static_conffiles() {
+	set --
+	sed -ne '/^[[:space:]]*$/d; /^#/d; p' \
+                /etc/sysupgrade.conf /lib/upgrade/keep.d/* 2>/dev/null |
+	{
+		while read -r filename; do
+			set -- "$@" "$filename"
+		done
+		find_bkpfiles "$@" 2>/dev/null
```

Comment:
find_bkpfiles is equivalent to the old find $(sed...) but it receives the filenames from args and $filter is included inside find_bkpfiles reading directly SKIP_UNCHANGED